LFPy/run_simulation.py

```diff
@@ -206,29 +233,40 @@ def _run_simulation_with_electrode(cell, cvode, electrode=None,
                                                                    rtfactor))
             t0 = time()
             ti = neuron.h.t
-    
-    try:
-        #calculate LFP after final fadvance()
-        i = 0
-        for sec in cell.allseclist:
-            for seg in sec:
-                imem[i] = seg.i_membrane_
-                i += 1
 
-        if rec_current_dipole_moment:
-            cell.current_dipole_moment[tstep, ] = np.dot(imem, midpoints)
-            
-        if to_memory:
-            for j, coeffs in enumerate(dotprodcoeffs):
-                electrodesLFP[j][:, tstep] = np.dot(coeffs, imem)
-        if to_file:
-            for j, coeffs in enumerate(dotprodcoeffs):
-                el_LFP_file['electrode{:03d}'.format(j)
-                            ][:, tstep] = np.dot(coeffs, imem)
+    # calculate LFP after final fadvance() if needed
+    if tstep < len(cell._neuron_tvec):
+        try:
+            i = 0
+            for sec in cell.allseclist:
+                for seg in sec:
+                    imem[i] = seg.i_membrane_
+                    i += 1
+
+            if rec_current_dipole_moment:
+                if not cvode.active():
+                    cell.current_dipole_moment[tstep, ] = np.dot(imem, midpoints)
+                else:
+                    cell.current_dipole_moment.append(np.dot(imem, midpoints))
```

Comment:
`np.hstack` instead of `append`?

Reply:
it actually needs to be hstack to keep the first dimension (n electrodes)